keras/src/backend/torch/export.py

```diff
@@ -1,35 +1,152 @@
-from keras.src import layers
+import copy
+import re
+import warnings
+
+import torch
+
+from keras.src import backend
+from keras.src import ops
 from keras.src import tree
+from keras.src.utils.module_utils import tensorflow as tf
+from keras.src.utils.module_utils import torch_xla
 
 
 class TorchExportArchive:
     def track(self, resource):
-        if not isinstance(resource, layers.Layer):
-            raise ValueError(
-                "Invalid resource type. Expected an instance of a "
-                "JAX-based Keras `Layer` or `Model`. "
-                f"Received instead an object of type '{type(resource)}'. "
-                f"Object received: {resource}"
-            )
+        raise NotImplementedError(
+            "`track` is not implemented in the torch backend. Use"
+            "`track_and_add_endpoint` instead."
+        )
+
+    def add_endpoint(self, name, fn, input_signature, **kwargs):
+        raise NotImplementedError(
+            "`add_endpoint` is not implemented in the torch backend. Use"
+            "`track_and_add_endpoint` instead."
+        )
+
+    def track_and_add_endpoint(self, name, resource, input_signature, **kwargs):
```

Comment:
Why was the new API needed? Can't we just make `track` and `add_endpoint` pure config ops, and make `write` do all the actual export logic?

Reply:
torch_xla relies on `ExportedProgram` generated by `torch.export.export` which requires a `torch.nn.Module`. As a result, it is difficult to decouple the logic into `track` and `add_endpoint` for torch_xla.

Additionally, the `fn` in `add_endpoint` is not applicable for torch_xla.

Reply:
It's certainly a bit confusing that `track`/`add_endpoint` only work in TF/JAX while `track_and_add_endpoint` only works with torch. But I see your point.

Reply:
`track_and_add_endpoint ` works across all backends. I have combined the logic of `track` and `add_endpoint` in the API for TF/JAX.